department-of-veterans-affairs/caseflow
# APPEALS-13122: Created models, migration, factories and updated specs

Resolves [APPEALS-13122](https://vajira.max.gov/browse/APPEALS-13122)

### Description
The purpose of this Pull request is to create new table (`membership_requests`), model, factory and write tests. 

### Acceptance Criteria
- [ ] Creation of a membership request class
- [ ] Creation of membership request factory
- [ ] Creation of migration of new membership request
- [ ] The membership request database entry will need to capture the following information:
  - [ ] Request ID
  - [ ] Requestor's User ID
  - [ ] Created at Date/Time
  - [ ] Status (Assigned/Approved/Denied/Cancelled)
  - [ ] Requesting Organization ID
  - [ ] Closed by User ID (nullable)
  - [ ] Closed at Date/Time

### Testing Plan
1. Do a git pull of this branch.
2. run `rake db:migrate`
3. Open rails console and check if the model is created or not by running the following command `MembershipRequest.count` It should not error out, it should return zero if there are no records
